lib/srv/app/cloud.go

```diff
@@ -170,12 +169,14 @@ func (c *cloud) getAWSSigninToken(ctx context.Context, req *AWSSigninRequest, en
 
 	// Sign In requests target IAM endpoints which don't require a region.
 	region := ""
-	session, err := c.cfg.SessionGetter(ctx, region, req.Integration)
+	baseCfg, err := c.cfg.AWSConfigProvider.GetConfig(ctx, region,
+		awsconfig.WithCredentialsMaybeIntegration(req.Integration),
+	)
 	if err != nil {
 		return "", trace.Wrap(err)
 	}
 
-	temporarySession, err := isSessionUsingTemporaryCredentials(session)
+	temporarySession, err := isSessionUsingTemporaryCredentials(ctx, baseCfg)
```

Comment:
do you mind adding a new log on the result of `isSessionUsingTemporaryCredentials`. it wasn't there before but it would really help debugging.